src/org/usfirst/frc/team2399/robot/RobotMap.java

```diff
@@ -9,67 +9,80 @@
-	public static final int DRIVETRAIN_LEFT_TALON_FRONT_ADDRESS = 4;
-	public static final int DRIVETRAIN_LEFT_BACK_TALON_ADDRESS = 3;
-	public static final int DRIVETRAIN_RIGHT_BACK_TALON_ADDRESS = 2;
-	public static final int CLIMBER_TALON_ADDRESS = 2399;
-
-	/**
-	 * Ports for sensors and joysticks
-	 */
 	public static final int JOYDRIVE_LEFT_STICK_PORT = 0;
 	public static final int JOYDRIVE_RIGHT_STICK_PORT = 1;
 
 	/**
-	 * Forward Constants To ensure positive values of motion are what you
+	 * FORWARD CONSTANTS
+	 * To ensure positive values of motion are what you
 	 * expect.
-	 *
+	 * TODO: Set actual forward constants for encoders
 	 */
 
 	public static final int JOYDRIVE_FORWARD_CONSTANT = -1;
 	public static final int DRIVETRAIN_FORWARD_LEFT_CONSTANT = 1;
 	public static final int DRIVETRAIN_FORWARD_RIGHT_CONSTANT = 1;
 	public static final int CLIMBER_FORWARD_CONSTANT = 1;
-
-	// For example to map the left and right motors, you could define the
-	// following variables to use with your drivetrain subsystem.
-	// public static int leftMotor = 1;
-	// public static int rightMotor = 2;
-
-	// If you are using multiple modules, make sure to define both the port
-	// number and the module. For example you with a rangefinder:
-	// public static int rangefinderPort = 1;
-	// public static int rangefinderModule = 1;
+	public static final int DRIVETRAIN_ENCODER_FORWARD_LEFT_CONSTANT = 1;
+	public static final int DRIVETRAIN_ENCODER_FORWARD_RIGHT_CONSTANT = 1;
 	
-	/**
-	 * Preset Speeds
-	 */
+	public static final boolean REVERSE_LEFT_FRONT_MOTOR_OUTPUT_CONSTANT = (DRIVETRAIN_FORWARD_LEFT_CONSTANT == -1);
+	public static final boolean REVERSE_RIGHT_FRONT_MOTOR_OUTPUT_CONSTANT = (DRIVETRAIN_FORWARD_LEFT_CONSTANT == -1);
+	public static final boolean REVERSE_LEFT_ENCODER_OUTPUT_CONSTANT = (DRIVETRAIN_ENCODER_FORWARD_LEFT_CONSTANT == -1);
+	public static final boolean REVERSE_RIGHT_ENCODER_OUTPUT_CONSTANT = (DRIVETRAIN_ENCODER_FORWARD_RIGHT_CONSTANT == -1);
 	
 	/**
+	 * PRESET SPEEDS
 	 * Preset speeds for climber to be used elsewhere in code
 	 * Currently at full speed - may need to be less depending on the mechanism
+	 * TODO: set actual speeds
 	 */
-	//TODO: set actual speeds
 	public static final double CLIMB_UP = 1.0;
 	public static final double CLIMB_DOWN = -1.0;
 	public static final double CLIMB_STOP = 0.0;
+	
+	/**
+	 * ENCODER CONSTANTS
+	 * TODO: Set actual values for constants
+	 */
+	
+	public static final double DISTANCE_DECREMENENT_CONSTANT = 0;
+	public static final double DISTANCE_INCREMENT_CONSTANT = 0;
+	public static final double ANGLE_P_CONSTANT = 0;
+	
+	/**
+	 * CALCULATION CONSTANTS
+	 */
+	
+	public static final double DRIVETRAIN_WHEEL_DIAMETER_CONSTANT = 0;
+	public static final double DRIVETRAIN_WHEEL_CIRCUMFERENCE_CONSTANT =  DRIVETRAIN_WHEEL_DIAMETER_CONSTANT * Math.PI;
+	
+	/**
+	 * Speed Soft Limits
+	 * TODO: Set velocity soft limits
+	 */
+	
+	public static final double VELOCITY_LOWER_SOFT_LIMIT = 0;
```

Comment:
You mention this in the todo, but we need to talk to Design to get these before this is ready for merge